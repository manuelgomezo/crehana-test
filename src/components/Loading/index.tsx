import styles from './styles.module.scss';

function Loading() {
  return (
    <div className={styles.ripple}>
      <div />
      <div />
    </div>
  );
}

export default Loading;
