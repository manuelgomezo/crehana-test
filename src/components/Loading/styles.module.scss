.ripple {
  align-items: center;
  display: flex;
  height: 80px;
  justify-content: center;
  margin: 20px auto;
  position: relative;
  text-align: center;

  div {
    animation: rippleAnim 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
    border: 4px solid var(--color-primary-main);
    border-radius: 50%;
    opacity: 1;
    position: absolute;
  }

  div:nth-child(2) {
    animation-delay: -0.5s;
  }
}

@keyframes rippleAnim {
  0% {
    height: 0;
    opacity: 1;
    width: 0;
  }

  100% {
    height: 72px;
    opacity: 0;
    width: 72px;
  }
}
