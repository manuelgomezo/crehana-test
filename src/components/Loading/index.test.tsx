import { render } from '@testing-library/react';

import Loading from './index';

describe('#Loading', () => {
  test('It render without errors', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
