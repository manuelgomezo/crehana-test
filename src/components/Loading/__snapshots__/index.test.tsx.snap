// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Loading It render without errors 1`] = `
<div>
  <div
    class="ripple"
  >
    <div />
    <div />
  </div>
</div>
`;
